### PR TITLE
Feat/filtering

### PR DESCRIPTION
**Pull request by SimranjitKaurMaan**
1. Added Filtering of the products

-  Filter by Price
- Filter by Artist
- Filter by Rating
- Sort by Price
product-filters

### CODE REVIEW

File: src/components/productListing/Products.js [hunk 1]
```diff
@@ -1,27 +1,29 @@
-import { useEffect , useState} from "react";
+import { useEffect } from "react";
 import { Link } from "react-router-dom";
 import { useCart } from "../../contexts/cart-context";
+import { useProductList } from "../../contexts/productlist-context";
 import { useWishlist } from "../../contexts/wishlist-context";
```
Comment (rahulcs754) [inline src/components/productListing/Products.js:5]: you can create index file in context and access all context using one file

**Review by rahulcs754 — APPROVED** [review]
The code looks good and clean.
just small suggestion

File: src/reducers/FilterUtils.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          (firstItem, secondItem) => secondItem.price - firstItem.price
+        );
+      default:
+        return items;
+    }
+  };
+
+export const handleFilterByRating = ( items , {rating}) =>  {
+    const filteredItems = items.filter(product => product.rating >= rating);
+    //console.log(filteredItems);
```
Comment (shuntzhere) [inline src/reducers/FilterUtils.js]: Do remove the console.logs before the final merge.

**Review by shuntzhere — APPROVED** [review]
Nice clean code. Great Descriptive Readme file.

File: src/reducers/FilterUtils.js (hunk 2 above)
Comment (SimranjitKaurMaan) [reply src/reducers/FilterUtils.js]: Yes thats what I planning to do. I need logs for debugging right now.

File: src/components/productListing/Products.js (hunk 1 above)
Comment (SimranjitKaurMaan) [reply src/components/productListing/Products.js:5]: Thanks for the suggestion. Currently I am not planning to create another file to organize the imports.

**Pull request merged**